metals/src/main/scala/scala/meta/internal/metals/Command.scala

```diff
@@ -100,6 +100,67 @@ case class ParametrizedCommand[T: ClassTag](
   }
 }
 
+case class FolderIdentifier(
+    val folder: String
+)
```

Comment:
this can be value class
```suggestion
case class FolderIdentifier(
    val folder: String
) extends Anyval
```

Value classes are [almost](https://docs.scala-lang.org/overviews/core/value-classes.html#when-allocation-is-necessary) zero cost abstraction in Scala.